### ISSUE TITLE
The prediction result of the same picture is different every time.

### ISSUE THREAD

**Issue by wojiaozhuyt** [matterport/Mask_RCNN#1146]
I have train on my own dataset, which are small pest. After training, when I detected, the prediction result of the same picture is different every time.
How should I solve the problem?

**Comment by wojiaozhuyt** [comment]
Also the mAP is very low.

**Comment by javierfs** [comment]
how did you show the mAP?

**Comment by wojiaozhuyt** [comment]
> how did you show the mAP?

Thank you for your replyT！ I show the mAP by using the code in the coco.py. And I think the problem that needs to be solved most is that the prediction result of the same picture is different every time.
Looking forward to your reply.
Thanks.

**Comment by wojiaozhuyt** [comment]
@javierfs

**Comment by ash1995** [comment]
I'm having the same issue. I trained the Mask RCNN on my own dataset of satellite images and tried to predict on some images. However, the model gives random mask and bounding box predictions.

**Comment by ash1995** [comment]
In my case, the problem was I wasn't loading the weights of some layers during inference time (via the exclude parameter) and they were getting randomly initialised everytime I loaded the weights. Just had to change the code from this 

`model.load_weights(model_path, by_name=True, exclude=["mrcnn_bbox_fc", "mrcnn_class_logits", "mrcnn_mask"])` 

to this in my inference file

`model.load_weights(model_path, by_name=True)`
